ydb/core/viewer/viewer.h

```diff
@@ -70,12 +73,68 @@ struct TRequestState {
         , TraceId(traceId)
     {}
 
-    const NMon::TEvHttpInfo* operator ->() const {
-        return Request;
+    TRequestState(const NHttp::TEvHttpProxy::TEvHttpIncomingRequest* request)
+        : Request(request)
+    {}
+
+    TRequestState(const NHttp::TEvHttpProxy::TEvHttpIncomingRequest* request, NWilson::TTraceId traceId)
+        : Request(request)
+        , TraceId(traceId)
+    {}
+
+    bool HasHeader(TStringBuf name) const {
+        if (auto* request = std::get_if<const NMon::TEvHttpInfo*>(&Request)) {
+            return (*request)->Request.GetHeaders().HasHeader(name);
+        }
+        if (auto* request = std::get_if<const NHttp::TEvHttpProxy::TEvHttpIncomingRequest*>(&Request)) {
+            return NHttp::THeaders((*request)->Request->Headers).Has(name);
+        }
+        return false;
+    }
+
+    TString GetHeader(TStringBuf name) const {
+        if (auto* request = std::get_if<const NMon::TEvHttpInfo*>(&Request)) {
```

Comment:
```
return std::visit(TOverloaded{
  [&](const NMon::TEvHttpInfo* ptr) { ... },
  [&](const ...) { ... }
}, Request);

```
would be much more readable and wouldn't create errors if one extends variant

Reply:
readable? hm.